src/hist1d.jl

```diff
@@ -67,24 +123,7 @@ end
 Create a `Hist1D` with given bin `edges` and vlaues from
 array. Weight for each value is assumed to be 1.
 """
-function Hist1D(A::AbstractVector, r::AbstractRange{T}) where T <: AbstractFloat
-    s = step(r)
-    start = first(r)
-    start2 = start + 0.5s
-    stop = last(r)
-    L = length(r) - 1
-    counts = zeros(Int, L)
-    @inbounds for idx in eachindex(A)
-        # skip overflow
-        i = A[idx]
-        c = ifelse(i > stop, 0, 1)
-        c = ifelse(i < start, 0, c)
-        id = round(Int, (i - start2) / s) + 1
-        counts[clamp(id, 1, L)] += c
-    end
-    return Hist1D(Histogram(r, counts))
-end
-function Hist1D(A::AbstractVector, r::AbstractRange{T}) where T <: Integer
+function Hist1D(A::AbstractVector, r::AbstractRange{T}) where T <: Union{AbstractFloat,Integer}
```

Comment:
How would you feel about modifying this whole function to
```julia
function Hist1D(A::AbstractVector, r::AbstractRange{T}) where T <: Union{AbstractFloat,Integer}
    h = Hist1D(Int; bins=r)
    unsafe_push!.(Ref(h), A)
    return h
end
```
in order to then have only one place where we implement the binning logic? Before, this method creates a hist from 1e6 numbers in 3.3-3.4ms, and after modifying it's about 3.9-4.0ms